frontend/views/containers/contributions/IncomeForm.vue

```diff
@@ -182,18 +182,27 @@ export default {
       return this.form.option === 'yes'
     },
     inputIncomeError () {
+      let error = ''
       if (!this.$v.form.income.hasLowIncome) {
-        return this.L('If your income is the same or higher than the group\'s mincome change your answer to "Yes, I do"')
+        error = this.L('If your income is the same or higher than the group\'s mincome change your answer to "Yes, I do"')
       } else if (this.$v.form.income.$error) {
-        return this.infoRequired
+        error = this.infoRequired
       }
+      return error
     },
     saveButtonText () {
       const verb = this.isFirstTime ? 'Add' : 'Save'
-
-      if (!this.form.option) { return this.L('{verb} details', { verb }) }
-      if (this.canPledge) { return this.L('{verb} pledged details', { verb }) }
-      if (this.needsIncome) { return this.L('{verb} income details', { verb }) }
+      let text = ''
+      if (!this.form.option) {
+        text = this.L('{verb} details', { verb })
+      } else {
+        if (this.canPledge) {
+          text = this.L('{verb} pledged details', { verb })
+        } else {
+          if (this.needsIncome) { text = this.L('{verb} income details', { verb }) }
+        }
+      }
```

Comment:
This is a very unwieldy if block that is more complicated than what we had before. There are probably several ways it can be simplified, but one obvious one is to make it a single level `if/else if` chain of 3 conditions.

```js
if () {
  ...
} else if () {
  ...
} else if () {
  ...
}
```

Is logically equivalent to what's above.

However, it is even better to avoid using if statements altogether (when possible). `if/else` blocks are ugly and they also slow things down because at a low-level they create a branch condition for the CPU. There was a fantastic blog post that showed examples of how to do this but I'm blanking on it right now.

One trick I know is to refactor the code to use a map (in javascript, an object), e.g. instead of:

```
let text = ''
if (thing === 'thing1') {
  text = 'something'
} else if (thing === 'thing2') {
  text = 'something else'
}
return text
```

Do:

```js
return myMap[thing]
```

More elegant, and way faster.